### PR TITLE
HashingHub

### PR DESCRIPTION
**Pull request by u-hubar**
- HashingHub contract
- HashingAlgorithm interface
- SHA256 contract implementation

### CODE REVIEW

**Review by zeroxbt — APPROVED** [review]

**Review by branarakic — CHANGES_REQUESTED** [review]

File: contracts/HashingHub.sol [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import { HashingAlgorithm } from "./interface/HashingAlgorithm.sol";
```
[the remark this answers is not in the document]
Comment (u-hubar) [reply contracts/HashingHub.sol:5]: Done

File: contracts/interface/HashingAlgorithm.sol [hunk 2]
```diff
@@ -0,0 +1,8 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+
+interface HashingAlgorithm {
```
[the remark this answers is not in the document]
Comment (u-hubar) [reply contracts/interface/HashingAlgorithm.sol:6]: Done

File: contracts/HashingHub.sol [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    {
+        HashingAlgorithm hashingAlgorithm = HashingAlgorithm(algorithms[hashingAlgorithmId]);
+        return hashingAlgorithm.hash(data);
+    }
+
+    function setContractAddress(uint8 hashingAlgorithmId, address hashingContractAddress)
+        public
+        onlyOwner
+    {
+        algorithms[hashingAlgorithmId] = hashingContractAddress;
```
[the remark this answers is not in the document]
Comment (u-hubar) [reply contracts/HashingHub.sol:25]: Done

File: contracts/hashing/sha256.sol [hunk 4]
```diff
@@ -0,0 +1,12 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity^0.8.0;
+
+
+import { HashingAlgorithm } from "../interface/HashingAlgorithm.sol";
+
```
[the remark this answers is not in the document]
Comment (u-hubar) [reply contracts/hashing/sha256.sol:7]: Done

**Pull request merged**